Updated version and changelog.md towards v0.7.0 (#841)
fortran-lang/stdlib: CHANGELOG.md

# Version 0.7.0

Full release notes available at [v0.7.0] tag.

[v0.7.0]: https://github.com/fortran-lang/stdlib/releases/tag/v0.7.0

- new module `stdlib_constants`
  [#800](https://github.com/fortran-lang/stdlib/pull/800)

>   - Many mathematical constants and most common physical (codata) constants

Reviewer: ```suggestion
  - Many mathematical constants and most common physical ([codata](https://codata.org)) constants
```

Changes to existing scripts and modules
  - changes in CI
    - Use of `fortran-setup` for GCC, Intel LLVM and Intel Classic
      [#834](https://github.com/fortran-lang/stdlib/pull/834)
  - change in module `stdlib_hashmaps`
    - Support of hash map key generic interfaces
      [#827](https://github.com/fortran-lang/stdlib/pull/827)
  - changes in module `stdlib_io`
    - Addition of a Fortran format specifier in `loadtxt`
      [#805](https://github.com/fortran-lang/stdlib/pull/805)
  - changes in module `stdlib_linalg`
    - Support of extended and quad precision checking 
      [#821](https://github.com/fortran-lang/stdlib/pull/821)
    - Several fixes
      [#815](https://github.com/fortran-lang/stdlib/pull/815)
      [#818](https://github.com/fortran-lang/stdlib/pull/818)
      [#826](https://github.com/fortran-lang/stdlib/pull/826)
      [#830](https://github.com/fortran-lang/stdlib/pull/830)
      [#836](https://github.com/fortran-lang/stdlib/pull/836)

>     - New procedures for Singular Value Decomposition: `eig`, `eigh`, `eigvals`, `eigvalsh`

Reviewer: ```suggestion
    - New procedures for Eigenvalues and Eigenvectors computation: `eig`, `eigh`, `eigvals`, `eigvalsh`
```

>       [#808](https://github.com/fortran-lang/stdlib/pull/808)

Author: ```suggestion
      [#816](https://github.com/fortran-lang/stdlib/pull/816)
```

>     - New procedures for Eigenvalues and Eigenvectors computation: `svd`, `svdvals`

Reviewer: ```suggestion
    - New procedures for Singular Value Decomposition: `svd`, `svdvals`
```

>       [#816](https://github.com/fortran-lang/stdlib/pull/816)

Author: ```suggestion
      [#808](https://github.com/fortran-lang/stdlib/pull/808)
```
